java/src/main/java/com/logicalclocks/hsfs/engine/SparkEngine.java

```diff
@@ -143,7 +142,20 @@ public String getCertKey() {
   }
 
   public Dataset<Row> sql(String query) {
-    return sparkSession.sql(query);
+    try {
+      return sparkSession.sql(query);
+    } catch (Exception e) {
+      if (e.getMessage().contains("Permission denied")) {
+        Pattern pattern = Pattern.compile("inode=\"/apps/hive/warehouse/(.*)?_featurestore\\.db\"");
+        Matcher matcher = pattern.matcher(e.getMessage());
+        if (matcher.find()) {
+          String featureStore = matcher.group(1);
+          throw new RuntimeException(String.format("Cannot access feature store '%s'. " +
+              "It is possible to request access from admins of '%s'.", featureStore, featureStore));
```

Comment:
We don't use "project admins" as a concept, we use "data owner". Let's be consistent here so that users don't get confused.